main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FaresV1FileValidator.java

```diff
@@ -0,0 +1,33 @@
+package org.mobilitydata.gtfsvalidator.validator;
+
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.notice.ConditionallyForbiddenFileNotice;
+import org.mobilitydata.gtfsvalidator.notice.MissingRequiredFileNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsFareAttributeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsFareRuleTableContainer;
+
+public class FaresV1FileValidator extends FileValidator {
+
+  private final GtfsFareAttributeTableContainer fareAttributeTable;
+  private final GtfsFareRuleTableContainer fareRuleTable;
+
+  @Inject
+  public FaresV1FileValidator(
+      GtfsFareAttributeTableContainer fareAttributeTable,
+      GtfsFareRuleTableContainer fareRuleTable) {
+    this.fareAttributeTable = fareAttributeTable;
+    this.fareRuleTable = fareRuleTable;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    if (!fareAttributeTable.isMissingFile() && fareRuleTable.isMissingFile()) {
+      noticeContainer.addValidationNotice(
+          new MissingRequiredFileNotice(fareRuleTable.gtfsFilename()));
```

Comment:
@isabelle-dr I reused `MissingRequiredFileNotice` here, but wondering if I should introduce a new `MissingConditionallyRequiredFileNotice` to match `ConditionallyForbiddenFileNotice`?

Reply:
Interesting! The most coherent from my perspective would be to do either:
1. Use `MissingRequiredFileNotice` and `FordiddenFileNotice` for everything. This would mean introducing `FordiddenFileNotice` in this PR instead.
2. Use `MissingRequiredFileNotice`, `MissingConditionallyRequiredFileNotice`, `FordiddenFileNotice`, `ConditionallyForbiddenFileNotice`. This would mean introducing the new `MissingConditionallyRequiredFileNotice` as you mentioned.

From a user perspective: there are often "conditions", even for fields that are tagged as Required (for example, `area_id` is required if the file is present).
If I had `missing_conditionally_required_file` and `missing_required_file` in the same validation report, I would be confused. Same thing for forbidden files, and for fields.

On another note, we have a different approach for **calendar.txt** and **calendar_dates.txt**: we have a custom rule ([`missing_calendar_and_calendar_date_files`](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#missing_calendar_and_calendar_date_files)), which I think is the easiest for the user because we can provide the documentation they need & they don't have to go in the spec.

For these reasons, I would say first approach, or custom rules like I described [here](https://github.com/MobilityData/gtfs-validator/pull/1228#pullrequestreview-1116059614).

Reply:
Ok, that all sounds reasonable to me.  For now, I went with option #1.  I agree that custom notices offer the most flexibility for linking to documentation, etc, but I'm wondering if there are ways we can get similar functionality without so many custom classes.  Or if nothing else, introduce something for automatically generating RULES.md from code, since updating it each time is tedious / error-prone :)